fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -128,7 +231,10 @@ where
             .deploy_contract_with_id(salt, storage_slots, &contract, &root, &id)
             .map_err(InterpreterError::from_io)?;
 
-        let remaining_gas = create.limit();
+        let remaining_gas = create
```

Comment:
Could we deduct the predicate gas before deploying the contract to save some extra computation in case it fails?